src/System.Private.CoreLib/shared/System/Threading/Tasks/AsyncCausalityTracer.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !FEATURE_COMINTEROP
```

Comment:
We prefer conditions in the .csproj files to end-to-end ifdefs.

Reply:
Yep, I was not sure how to best indicate this file is for FEATURE_COMINTEROP only

Reply:
The condition for the file can be `'$(FeatureAsyncCausalityTracer)'!='true'` so that each runtime can have its own idea whether and when to use the noop implementation.

Reply:
@jkotas Where would I set that flag for CoreCLR FEATURE_COMINTEROP ?

Reply:
Adding this to System.Private.CoreLib.csproj in CoreCLR should do the trick:
```
  <PropertyGroup>
    <FeatureAsyncCausalityTracer Condition="'$(FeatureCominterop)' == 'true'">true</FeatureAsyncCausalityTracer>
  </PropertyGroup>
```

Reply:
ok